### ISSUE TITLE
Crash trap with custom logger and sys.stdout.encoding

### ISSUE THREAD

**Issue by KeyWeeUsr** [kivy/python-for-android#1018]
Start.c filed override `sys.stdout`, which has some properties set such as `encoding`. This opens a trap for anyone who will call `sys.stdout.encoding`, which I think is used even in some parts of the standard library.

A potential fix is [here](https://github.com/kivy/kivy/pull/5056), but that's only from Kivy's side and only applies for SDL2, therefore affected parts of p4a are:

* pygame
* webview
* service most likely too
* python only packages

An ugly fix might be to manually put `encoding = 'utf8'` into `LogFile`, but better one might be to try using the `sys.stdout` if there already is one and move its dictionary (variables) to our custom `LogFile` class and then apply our stuff. Then there'll be no such case of `sys.stdout.<random not available property>` won't be our fault and won't present a potential risk of crashing the whole application on some basic mistake e.g. while using a standard Python library.

**Comment by inclement** [comment]
Copying the stdout.encoding in particular might be a bad idea as the Android logger output might not actually use the same encoding. I'm not sure if this would matter, but it probably needs some thought.

**Comment by llfkj** [comment]
As to whether the LogFile class needs to have an `encoding` attribute, I don't know if there is an easy answer. 

In the python2 docs, they claim this:

>  stdout and stderr needn’t be built-in file objects: **any object is acceptable as long as it has a write() method that takes a string argument**. (Changing these objects doesn’t affect the standard I/O streams of processes executed by os.popen(), os.system() or the exec*() family of functions in the os module.)

In python3, they make no such claim that implementing `write()` is sufficient. (and indeed, their `sys.displayhook` example uses `sys.stdio.encoding` without testing if it exists). Here's what the docs say there:
> File objects used by the interpreter for standard input, output and errors:
> ...
> These streams are regular text files like those returned by the open() function. 

So I don't know what the safe thing to do is. From a python3 perspective, it looks like `sys.stdout` should be implemented like a full-on file object. Even then, in the py3 documentation for file objects, it claims that:
> The [encoding] attribute is read-only and may not be present on all file-like objects. It may also be `None`, in which case the file uses the system default encoding for converting Unicode strings.

In short, I think the latest version of [my patch](https://github.com/kivy/kivy/pull/5056) is the "right" way to query the attribute. Technically, LogFile does nothing wrong by not implementing `encoding`, though it is a bit of a trap (judging by examples on StackOverflow, and in the `sys.displayhook` section of the py3 docs.

**Comment by inclement** [comment]
@llfkj Thanks again for the patch. There's probably a case for adding this attribute in python-for-android, but since it seems our behaviour is technically okay, and I don't think it's ever been a problem before, I'll close this for now.
